src/filesystem_helper.cpp

```diff
@@ -474,6 +475,10 @@ bool remove_all(const path & p)
   return 0 == ret && false == file_options.fAnyOperationsAborted;
 #else
   DIR * dir = opendir(p.string().c_str());
+  if (dir == nullptr) {
+    return false;
+  }
+  RCPPUTILS_SCOPE_EXIT(closedir(dir); );
```

Comment:
I think you can just do:

```suggestion
  RCPPUTILS_SCOPE_EXIT(closedir(dir));
```

the macro already adds a `;` after the code.